cargo-dist/src/tasks.rs

```diff
@@ -2280,16 +2280,22 @@ pub fn gather_work(cfg: &Config) -> Result<DistGraph> {
     Ok(graph.inner)
 }
 
-/// See if we should dist this package
+/// See if we should dist this package.
 ///
-/// Some(disabled_reason) is returned if it shouldn't be
+/// Some(disabled_reason) is returned if it shouldn't be.
+///
+/// This code is written to assume a package and its binaries should be distable,
+/// and then runs through a battery of disqualifying reasons.
+///
+/// A notable consequence of this is that if --tag wasn't passed, then we will default to
```

Comment:
(caveat: i did not yet cross check our publish facing docs) it might be useful to also doc this behavior for end users?

Reply:
yeah so the inference system exists to enable a few usecases:

* greasing the wheels on someone just messing around locally and trying cargo-dist out
* pr checks don't have a tag to use so need to use a fake one and "whatever the current version is" works great for that

definitely worth calling out in the pr-check docs if it isn't already